organization/_posts/2018-05-02-packaging.md

```diff
@@ -0,0 +1,89 @@
+
+[Introduction - Graeme](https://indico.cern.ch/event/719557/contributions/2957742/attachments/1626928/2624333/HSF_Packaging_Group_Intro_2018-05-02.pdf)
+===========================
+- [Guix presentation tomorrow](https://indico.cern.ch/event/719851/)
+  - GNU reimplementation of Nix?
+    - Chris thinks it may be a fork from Nix in the past
+    - No non-GPL software allowed?
+  - Graeme will attend
+- [UK RSE](http://rse.ac.uk) - [2018 conference in Birmingham 3-4th September](http://rse.ac.uk/conf2018/)
+  - Abstracts submitted: one on HSF generally, and one on Packaging
+  - Will hear if accepted by end of May
+- Moving onto today’s agenda, as ever, contact Graeme and Ben if you have contributions, updates for future meetings
+
+[Support for multiple micro-architectures - Pere](https://indico.cern.ch/event/719557/contributions/2965980/attachments/1642767/2624258/HSF-Packaging-20180502.pdf)
+===============================================
+- Need separate stacks for instruction sets (e.g fma, avx etc)
+  - How to make available on CVMFS?
+  - Select stack at runtime from CPU detection
+  - Not all packages need/use these specialist instructions
+- Implementing this in LCGMake - recap given on recipes and views
+- Naming platforms - packages tagged on `arch-os-compiler-optimization`
+  - Extend arch with `arch+iset1+iset2+...+isetN` to describe which instruction sets
+    the package was compiled with
+- In LCGMake, new argument to package to select full instruction set build
+  (via compiler wrapper), which adds platform tag extension
+  - Some issues with static libraries and dependencies
+- Packages not using this are placed under arch-only tag
+- Setting up the environment - use views to allow transparent set of instruction set dependent packages combined with those that don’t require specific instruction sets.
+- System now in use is minimalistic and only builds a subset of packages
+- Is working, and used for GeantV nightlies.
+- As we have to define built instruction sets, need to define the default/minimal we can expect
+- Still need a tool to discover runtime discovery and selection of "best" match.
+- Question: How to manage the dependencies between packages?
+  - Change of a low lying library - do you need to build all of the dependencies to be really sure (e.g., inlining)
+- Building ROOT is problematic - dictionaries can be sensitive to micro-architecture.
+- For "bare" packages, does choice of instruction set/optimization level influence interface?
+- HSF note was rather blocked by these dependency issues
+- What about fat binaries, that support multiple micro-architectures in one library/binary?
+  - Larger binaries and cost of runtime switching - though these are not well quantified for our environment
+  - LHCb had reproducibility issues with fat libraries
```

Comment:
In addition, adding support to new micro-actitectures later will need to modify long time released software, which is a serious risk in reproducibility.